x-pack/plugins/fleet/public/applications/fleet/app.tsx

```diff
@@ -0,0 +1,257 @@
+                  ) : (
+                    <FormattedMessage
+                      id="xpack.fleet.securityRequiredErrorTitle"
+                      defaultMessage="Security is not enabled"
+                    />
+                  )}
+                </h2>
+              }
+              body={
+                <p>
+                  {permissionsError === 'MISSING_SUPERUSER_ROLE' ? (
+                    <FormattedMessage
+                      id="xpack.fleet.permissionDeniedErrorMessage"
+                      defaultMessage="You are not authorized to access Fleet. Fleet requires {roleName} privileges."
+                      values={{ roleName: <EuiCode>superuser</EuiCode> }}
+                    />
+                  ) : (
+                    <FormattedMessage
+                      id="xpack.fleet.securityRequiredErrorMessage"
+                      defaultMessage="You must enable security in Kibana and Elasticsearch to use Fleet."
+                    />
+                  )}
+                </p>
+              }
+            />
+          </Panel>
+        )}
+      </ErrorLayout>
+    );
+  }
+
+  if (!isInitialized || initializationError) {
+    return (
+      <ErrorLayout>
+        {initializationError ? (
+          <Error
+            title={
+              <FormattedMessage
+                id="xpack.fleet.initializationErrorMessageTitle"
+                defaultMessage="Unable to initialize Fleet"
+              />
+            }
+            error={initializationError}
+          />
+        ) : (
+          <Loading />
+        )}
+      </ErrorLayout>
+    );
+  }
+
+  return <>{children}</>;
+});
+
+/**
+ * Fleet Application context all the way down to the Router, but with no permissions or setup checks
+ * and no routes defined
+ */
+export const FleetAppContext: React.FC<{
+  basepath: string;
+  coreStart: CoreStart;
+  setupDeps: IngestManagerSetupDeps;
+  startDeps: IngestManagerStartDeps;
+  config: IngestManagerConfigType;
+  history: AppMountParameters['history'];
+  kibanaVersion: string;
+  extensions: UIExtensionsStorage;
+}> = memo(
+  ({ children, coreStart, setupDeps, startDeps, config, history, kibanaVersion, extensions }) => {
+    const isDarkMode = useObservable<boolean>(coreStart.uiSettings.get$('theme:darkMode'));
+
+    return (
+      <coreStart.i18n.Context>
+        <KibanaContextProvider services={{ ...coreStart }}>
+          <EuiErrorBoundary>
+            <DepsContext.Provider value={{ setup: setupDeps, start: startDeps }}>
+              <ConfigContext.Provider value={config}>
+                <KibanaVersionContext.Provider value={kibanaVersion}>
+                  <EuiThemeProvider darkMode={isDarkMode}>
+                    <UIExtensionsContext.Provider value={extensions}>
+                      <FleetStatusProvider>
+                        <IntraAppStateProvider kibanaScopedHistory={history}>
+                          <PackageInstallProvider notifications={coreStart.notifications}>
```

Comment:
Can you note why `PackageInstallProvider` was moved outside `Router`? I realize this is still Draft and don't ask because I'm against it. Just add a note to highlight and explain the goal when it's ready for review.

I haven't thought about Router & Context recently enough to know what might be affected or potential consequences

Reply:
actually - now that I look at this again (the refactored version), I don't think it matters. In order to avoid any potential issues, I will move it back inside of `<Router>`.  Thanks @jfsiii